asteroid/foo.py

```diff
@@ -0,0 +1,162 @@
+import os
+import torch
+import numpy as np
+import warnings
+
+try:
+    from typing import Protocol
```

Comment:
Protocols are a new feature in Python 3.8 that define an interface without requiring the implementing classes to inherit some abstract base class (ie. duck typing). It is used only for documentation right now